spec/services/preserved_object_handler_create_spec.rb

```diff
@@ -87,4 +86,126 @@
+          endpoint.endpoint_type = Endpoint.default_storage_root_endpoint_type
+          endpoint.endpoint_node = Settings.endpoints.storage_root_defaults.endpoint_node
+          endpoint.storage_location = storage_dir
+          endpoint.recovery_cost = Settings.endpoints.storage_root_defaults.recovery_cost
+          # endpoint.preservation_policies = PreservationPolicy.default_preservation_policy
+        end
+      end
+
+      it 'creates preserved object and preserved copy object with status "invalid_moab"' do
+        po_args = {
+          druid: invalid_druid,
+          current_version: incoming_version,
+          preservation_policy: PreservationPolicy.default_preservation_policy
+        }
+        pc_args = {
+          preserved_object: an_instance_of(PreservedObject), # TODO: see if we got the preserved object that we expected
+          version: incoming_version,
+          size: incoming_size,
+          endpoint: ep,
+          status: Status.invalid
+        }
+
+        expect(PreservedObject).to receive(:create!).with(po_args).and_call_original
+        expect(PreservedCopy).to receive(:create!).with(pc_args).and_call_original
+        po_handler = described_class.new(invalid_druid, incoming_version, incoming_size, ep)
+        po_handler.create_with_validation
+      end
+
+      context 'db update error' do
+        context 'ActiveRecordError' do
+          let(:result_code) { PreservedObjectHandler::DB_UPDATE_FAILED }
+          let(:results) do
+            allow(Rails.logger).to receive(:log)
+            # FIXME: couldn't figure out how to put next line into its own test
+            expect(Rails.logger).to receive(:log).with(Logger::ERROR, /#{db_update_failed_prefix_regex_escaped}/)
+
+            po = instance_double("PreservedObject")
+            allow(PreservedObject).to receive(:create!).with(hash_including(druid: invalid_druid))
+                                                       .and_raise(ActiveRecord::ActiveRecordError, 'foo')
+            allow(po).to receive(:destroy) # for after() cleanup calls
+            po_handler = described_class.new(invalid_druid, incoming_version, incoming_size, ep)
+            po_handler.create_with_validation
+          end
+
+          context 'DB_UPDATE_FAILED error' do
+            it 'prefix' do
+              expect(results).to include(a_hash_including(result_code => a_string_matching(db_update_failed_prefix_regex_escaped)))
+            end
+            it 'specific exception raised' do
+              expect(results).to include(a_hash_including(result_code => a_string_matching('ActiveRecord::ActiveRecordError')))
+            end
+            it "exception's message" do
+              expect(results).to include(a_hash_including(result_code => a_string_matching('foo')))
+            end
+          end
+        end
+      end
+    end
+
+    context 'returns' do
+      let!(:result) { po_handler.create }
```

Comment:
was the intention to test `create` here?  i'd have expected `.create_with_validation`?  (since previously written tests should cover the refactoring of `.create`, right?)

Reply:
oops ya copy and paste left over lol